### PR TITLE
Add parameter server functionality

### PR DESCRIPTION
**Pull request by Acuadros95**
Before merge:
- [ ] Decide if we prefer `rclc_parameter_set` and `rclc_parameter_get` API or `rclc_parameter_set_[type]` and `rclc_parameter_get_[type]`. Check [here](https://github.com/ros2/rclc/blob/50cb17e7fe3e0f2a3aae4bed78061bb176a1d1c9/rclc_parameter/include/rclc_parameter/rclc_parameter.h#L133). Could you please give your opinions @JanStaschulat @ralph-lange @jamoralp @Acuadros95?

Related: 
- https://github.com/micro-ROS/rosidl_typesupport_microxrcedds/pull/27
- https://github.com/ros2/rosidl/pull/584

### CODE REVIEW

**Review by pablogs9 — COMMENTED** [review]
In general:
- We need a user callback that gets triggered when a parameter is modified remotely. A callback that is triggered when set parameter service is called.
- We need to notify when both the user o the remote operator, modifies any parameter.

File: rclc_parameter/src/rclc_parameter/parameter_server.c [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+
+rcl_ret_t rclc_parameter_server_init_default(
+        rcl_parameter_server_t* param_server,
+        rcl_node_t* node)
+{
+    // TODO: Add ret check
+    rcl_ret_t ret;
+
+    param_server->param_number = 0;
+    rcl_interfaces__msg__Parameter__Sequence__init(&param_server->parameter_list, RCLC_UXRCE_MAX_PARAMETERS);
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply rclc_parameter/src/rclc_parameter/parameter_server.c:145]: Are we sure that `rcl_interfaces__msg__Parameter__Sequence__init` inits correctly the struct? Verify that internal sequences are being correctly inited.

File: rclc_parameter/include/rclc_parameter/parameter_server.h [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        rcl_node_t* node,
+        char* service_name,
+        const rosidl_service_type_support_t* srv_type);
+
+RCL_PUBLIC
+rcl_ret_t
+rclc_add_parameter_bool(
+        rcl_parameter_server_t* param_server,
+        const char* parameter_name,
+        bool value);
```
[the remark this answers is not in the document]
Comment (Acuadros95) [reply rclc_parameter/include/rclc_parameter/parameter_server.h]: Updated function: https://github.com/ros2/rclc/blob/ff6de4acafa361be8a3969336cf7ac0d9ee7eb08/rclc_parameter/include/rclc_parameter/rclc_parameter.h#L84-L87

File: rclc_parameter/src/rclc_parameter/parameter_server.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (Acuadros95) [reply rclc_parameter/src/rclc_parameter/parameter_server.c]: `RCLC_UXRCE_MAX_PARAMETERS` removed, parameter number is now configured on the parameter server initialization

https://github.com/ros2/rclc/blob/ff6de4acafa361be8a3969336cf7ac0d9ee7eb08/rclc_parameter/include/rclc_parameter/rclc_parameter.h#L70-L73

Also, should the maximum string length be configurable on build time? Its value is fixed right now:
https://github.com/ros2/rclc/blob/ff6de4acafa361be8a3969336cf7ac0d9ee7eb08/rclc_parameter/src/rcl_interfaces/string/string_functions.c#L24

File: rclc_parameter/include/rclc_parameter/parameter_server.h [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+RCL_PUBLIC rcl_ret_t rclc_parameter_server_fini(
+        rcl_parameter_server_t* param_server,
+        rcl_node_t* node);
+
+RCL_PUBLIC rcl_ret_t rclc_executor_add_parameter_server(
+        rclc_executor_t* executor,
+        rcl_parameter_server_t* param_server);
+
+rcl_ret_t rclc_parameter_server_init_service(
```
[the remark this answers is not in the document]
Comment (Acuadros95) [reply rclc_parameter/include/rclc_parameter/parameter_server.h]: Added public API header: [include/rclc_parameter/rclc_parameter](https://github.com/ros2/rclc/blob/ff6de4acafa361be8a3969336cf7ac0d9ee7eb08/rclc_parameter/include/rclc_parameter/rclc_parameter.h)

**Comment by pablogs9** [comment]
Hello @JanStaschulat do you know how can `cpplint` can fail with

```
00:01:06.727 3: /tmp/ws/src/rclc/rclc_parameter/include/rclc_parameter/config.h:15:  #ifndef header guard has wrong style, please use: RCLC_PARAMETER__CONFIG_H_  [build/header_guard] [5]
```

if `rclc_parameter/config.h` does not exists? Also, in my local a `colcon test` does no report any failure... Any idea?

And on the other hand, is the github action `CI RCLC Rolling / build (ubuntu-20.04, rolling) (pull_request)` working properly?

**Comment by pablogs9** [comment]
Moved to https://github.com/ros2/rclc/pull/88

**Pull request closed without merging**